presto-parquet/src/main/java/com/facebook/presto/parquet/reader/ParquetReader.java

```diff
@@ -627,7 +633,24 @@ private static Block rewriteLongArrayBlock(LongArrayBlock longArrayBlock, Type t
                 newBlockBuilder.appendNull();
             }
             else {
-                targetType.writeLong(newBlockBuilder, longArrayBlock.getLong(position, 0));
+                targetType.writeLong(newBlockBuilder, longArrayBlock.getLong(position));
+            }
+        }
+
+        return newBlockBuilder.build();
+    }
+
+    private static Block rewriteTimeLongArrayBlock(LongArrayBlock longArrayBlock, Type targetType)
+    {
+        int positionCount = longArrayBlock.getPositionCount();
+        BlockBuilder newBlockBuilder = targetType.createBlockBuilder(null, positionCount);
+        for (int position = 0; position < positionCount; position++) {
+            if (longArrayBlock.isNull(position)) {
+                newBlockBuilder.appendNull();
+            }
+            else {
+                // We shift the bits to encode timezone information
+                targetType.writeLong(newBlockBuilder, longArrayBlock.getLong(position) << 12);
```

Comment:
I have the same question here as well, is it possible to use the utility methods present here: https://github.com/prestodb/presto/blob/5e1f28f0a3046f2dd64cbe0c5f639b862b1df02a/presto-common/src/main/java/com/facebook/presto/common/type/DateTimeEncoding.java